monkestation/code/modules/clothing/suits/reactive_armour.dm

```diff
@@ -0,0 +1,300 @@
+	var/list/current_herd = list()
+	var/list/pet_type_cache = list()
+	reactivearmor_cooldown_duration = 90 SECONDS
+
+/obj/item/clothing/suit/armor/reactive/herd/Initialize(mapload, new_lifespan)
+	. = ..()
+	//Copied from the petsplosion anomaly since it's already figured out
+	pet_type_cache = subtypesof(/mob/living/simple_animal/pet)
+	pet_type_cache -= list(/mob/living/simple_animal/pet/penguin,
+		/mob/living/simple_animal/pet/dog/corgi/narsie,
+		/mob/living/simple_animal/pet/gondola/gondolapod,
+		/mob/living/simple_animal/pet/gondola,
+		/mob/living/simple_animal/pet/dog)
+
+	pet_type_cache += list(/mob/living/simple_animal/cow,
+		/mob/living/simple_animal/sloth,
+		/mob/living/simple_animal/mouse,
+		/mob/living/simple_animal/parrot,
+		/mob/living/simple_animal/chicken,
+		/mob/living/simple_animal/cockroach,
+		/mob/living/simple_animal/crab)
+
+/obj/item/clothing/suit/armor/reactive/herd/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(!active)
+		return FALSE
+	if(prob(hit_reaction_chance))
+		if(world.time < reactivearmor_cooldown)
+			owner.visible_message("<span class='danger'>The animal generator is still recharging!</span>")
+			return FALSE
+		playsound(get_turf(src),'sound/effects/empulse.ogg', 100, 1)
+		owner.visible_message("<span class='danger'>[src] blocks [attack_text], and summons a herd of animals!</span>")
+		var/turf/owner_turf = get_turf(owner)
+		become_animal(owner)
+		for(var/i in 1 to rand(7,10)) //Summon the disguise herd
+			var/mob/living/simple_animal/pet/new_animal = pick(pet_type_cache)
+			new_animal = new new_animal(owner_turf)
+			current_herd += new_animal
+
+		reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
+		return TRUE
+
+/obj/item/clothing/suit/armor/reactive/herd/proc/become_animal(mob/user)
+	if(!ishuman(user))
+		return
+	var/mob/living/simple_animal/pet/chosen_animal = pick(pet_type_cache)
+	chosen_animal = new chosen_animal(get_turf(user))
+	user.forceMove(chosen_animal)
+	user.mind.transfer_to(chosen_animal)
+	ADD_TRAIT(user, TRAIT_NOBREATH, type) //so they dont suffocate while inside the animal
+	addtimer(CALLBACK(src, .proc/end_herd, chosen_animal), 30 SECONDS)
+
+/obj/item/clothing/suit/armor/reactive/herd/proc/end_herd(mob/living/affected_mob)
+	var/mob/living/carbon/human/human_mob = locate() in affected_mob
+	affected_mob.mind.transfer_to(human_mob)
+	human_mob.grab_ghost()
+	human_mob.forceMove(get_turf(affected_mob))
+	REMOVE_TRAIT(human_mob, TRAIT_NOBREATH, type)
+	qdel(affected_mob)
+
+	for(var/mob/living/herd_animal in current_herd)
+		current_herd -= herd_animal
+		qdel(herd_animal)
+
+//Fluidic Armour
+/obj/item/clothing/suit/armor/reactive/wet
+	name = "reactive wet armor"
+	desc = "An experimental suit of armor that's a little more damp than usual."
+	reactivearmor_cooldown_duration = 2 MINUTES
+	var/static/list/random_liquid_list = list(
```

Comment:
This honestly seems a bit weak for how rare some of these are.
I'd say just have it take a random safe chem like the non-dangerous fluidic anomaly.